src/Database/Transaction.php

```diff
@@ -94,7 +82,7 @@ public function run(callable $callback): void
      */
     protected function shouldBeginTransaction(): bool
     {
-        return !$this->open && (false === is_int($this->size) || 0 === $this->size || 1 === $this->count);
+        return !$this->open && (0 === $this->size || 1 === $this->count);
```

Comment:
Eventually for visual peope like me that are bad at reading inline logical :)
```suggestion
        return (
            false === $this->open
            && (
                0 === $this->size
                || 1 === $this->count
            )
        );
```

Reply:
Hmmm - I'm not even 100% sure that code is doing what what is expected. At least based on tests, nothing changes if we make it 

`return !$this->open && 1 === $this->count;`

So either tests are not actually covering that functionality or some of that code is superfluous.

I'm going to leave that whole question for a different issue.

Reply:
I created #76 so we don't lose this as a genuine concern.